From f7df29f7dc7ef78ffd34cfd13f7ccaae9a9f421c Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Tue, 27 May 2025 18:46:27 +0800
Subject: [PATCH 090/123] mtk: mt76: mt7990: add support critical packet mode
 for mt7990

For mt7990 chipsets, critical packet mode shall be enable. Without this
patch, some higher priority packets may be put in the wrong AC queue.

Without this patch, the default setting of TXS_CTRL CR is 3, which means
priority packet will be put into AC BK queue. This causes the MCS could
not stand on MCS9 when using BCC coding type to transmit to the TGBE
Intel BE200 Linux station.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>

---
 mt7996/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mt7996/main.c b/mt7996/main.c
index 68ca9974..dc558dd7 100644
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -100,7 +100,7 @@ static int mt7996_start(struct ieee80211_hw *hw)
 
 	mutex_lock(&dev->mt76.mutex);
 	ret = mt7996_mcu_set_hdr_trans(dev, true);
-	if (!ret && is_mt7992(&dev->mt76)) {
+	if (!ret && !is_mt7996(&dev->mt76)) {
 		u8 queue = mt76_connac_lmac_mapping(IEEE80211_AC_VI);
 
 		ret = mt7996_mcu_cp_support(dev, queue);
-- 
2.45.2

